### ISSUE TITLE
Multiple alpha build versions' IDL fails on first MDSconnect

### ISSUE THREAD

**Issue by ModestMC** [MDSplus/mdsplus#2580]
**Affiliation**
General Atomics / DIII-D CSS

**Version(s) Affected**
Alpha Releases 7.139.23, 7.139.28, 7.139.35-37 

**Platform**
RHEL 8

**Describe the bug**
Currently on our systems we use  alpha_release-7.130.1, but we are needing to update our build to some version which includes the fixes made in #2537. We have now tested several versions of MDSplus to try and find a working build, however each version we have tested has a bug where the first IDL mdsconnect attempt errors out. Furthermore the error occurs each time a new IDL shell is created (see screenshot below).

Also, it appears the status variable doesn't catch anything either:
```
IDL> mdsconnect,'[the server]',status=status
% Compiled module: MDSCONNECT.
% Compiled module: MDSDISCONNECT.
% MDSCONNECT: Error connecting to [the server]
% Execution halted at: $MAIN$
IDL> help,status
STATUS          UNDEFINED = <Undefined>
IDL> mdsconnect,'[the server]',status=status
IDL> help,status
STATUS          INT       =        1
```

**To Reproduce**
``` git checkout [the particular tag]
./bootstrap
./configure --prefix=[your build directory]
make
make install
[whatever you do for pathing]
idl
>mdsconnect, '[the URL]
```

**Expected behavior**
It should just connect with no error the way it does on the second attempt.

**Screenshots**
![Screenshot 2023-05-30 at 2 49 39 PM](https://github.com/MDSplus/mdsplus/assets/45748401/42a2445f-f748-4f55-8400-7088f797d244)

**Additional context**
We just need a new version of MDSplus that fixes the numpy string issue/update that works otherwise.

**Comment by joshStillerman** [comment]
I am pretty sure we have a fix for this. Will investigate tomorrow. Sent from my iPhoneOn May 30, 2023, at 7:35 PM, Mitchell Clark ***@***.***> wrote:﻿
Affiliation
General Atomics / DIII-D CSS
Version(s) Affected
Alpha Releases 7.139.23, 7.139.28, 7.139.35-37
Platform
RHEL 8
Describe the bug
Currently on our systems we use  alpha_release-7.130.1, but we are needing to update our build to some version which includes the fixes made in #2537. We have now tested several versions of MDSplus to try and find a working build, however each version we have tested has a bug where the first IDL mdsconnect attempt errors out. Furthermore the error occurs each time a new IDL shell is created (see screenshot below).
Also, it appears the status variable doesn't catch anything either:
IDL> mdsconnect,'[the server]',status=status
% Compiled module: MDSCONNECT.
% Compiled module: MDSDISCONNECT.
% MDSCONNECT: Error connecting to [the server]
% Execution halted at: $MAIN$
IDL> help,status
STATUS          UNDEFINED = <Undefined>
IDL> mdsconnect,'[the server]',status=status
IDL> help,status
STATUS          INT       =        1

To Reproduce
./bootstrap
./configure --prefix=[your build directory]
make
make install
[whatever you do for pathing]
idl
>mdsconnect, '[the URL]

Expected behavior
It should just connect with no error the way it does on the second attempt.
Screenshots

Additional context
We just need a new version of MDSplus that fixes the numpy string issue/update that works otherwise.
